core/healthmon/healthmon.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Health Monitor
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/rb.h>
+#include <sys/threads.h>
+
+#define ARG_SEPARATOR '@'
+
+typedef struct {
+	rbnode_t node;
+	char **argv;
+	pid_t pid;
+} proc_t;
+
+
+static struct {
+	rbtree_t ptree;
+} common;
+
+
+static int ptree_compare(rbnode_t *n1, rbnode_t *n2)
+{
+	proc_t *p1 = lib_treeof(proc_t, node, n1);
+	proc_t *p2 = lib_treeof(proc_t, node, n2);
+
+	if (p1->pid < p2->pid)
+		return -1;
+
+	else if (p1->pid > p2->pid)
+		return 1;
+
+	return 0;
+}
+
+
+static void usage(const char *progname)
+{
+	printf("Phoenix-RTOS Health Monitor - process respawner\n");
+	printf("Usage: %s progname1[" "%c" "arg1" "%c" "arg2" "%c" "...] [progname2...]\n", progname,
+		ARG_SEPARATOR, ARG_SEPARATOR, ARG_SEPARATOR);
+}
+
+
+static int spawn(proc_t *p)
+{
+	pid_t pid;
+
+	pid = spawnSyspage(NULL, p->argv[0], p->argv);
+	if (pid < 0)
+		return -ENOEXEC;
```

Comment:
`spawnSyspage` returns also `ENOENT`, `EINVAL` and others. 

```suggestion
		return pid;
```

Reply:
Done